op-challenger/game/keccak/scheduler.go

```diff
@@ -81,9 +90,13 @@ func (s *LargePreimageScheduler) verifyOraclePreimages(ctx context.Context, orac
 	if err != nil {
 		return err
 	}
+	period, err := oracle.ChallengePeriod(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to load challenge period: %w", err)
+	}
```

Comment:
It feels like this is very redundant as well since the challenge period is immutable, so once it's fetched from the preimage oracle, we should just be able to use it. So instead of pointing to oracles, we could use oracle fetchers that also cache the challenge period - similar to the pattern we use in the trace accessor whereby the provider is created if it's never been routed to iirc. Just a thought - not blocking, but maybe worth a ticket if this is a good optimization.

Reply:
It's probably overkill since this is such a simple call and only called once per update cycle (I'd be slightly more worried if it called once per preimage), but I added caching for it in the oracle bindings which I think is a useful pattern to use.  The contract bindings know the value is immutable so can safely cache but the rest of the code continues to make no assumptions about it.